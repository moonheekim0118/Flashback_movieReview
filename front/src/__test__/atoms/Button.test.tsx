import 'jsdom-global/register'; 
import React from "react";
import { mount } from 'enzyme';
import Button from '../../atoms/Buttons';

describe('<Button/>', () => {
   let container = null;
   const mockFn = jest.fn();

   it('renders correctly', ()=>{ // 렌더링 테스트 
      container=mount(<Button title="test" fill={false} shadow={false} disabled={false}  onClick={mockFn} color="lightPurple"/>);
   });

   it('matches snapshot', ()=>{ // 스냅샷 비교 
    expect(container.html()).toMatchSnapshot();
   });

   it('should call onClick if button is clicked',()=>{ // onClick 버튼 시뮬레이션 
      container.simulate('click');
      expect(mockFn).toHaveBeenCalled();
   });

   
})
