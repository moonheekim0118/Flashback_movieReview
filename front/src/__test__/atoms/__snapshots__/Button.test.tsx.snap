// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> matches snapshot 1`] = `"<button fill=\\"false\\" color=\\"lightPurple\\" class=\\"Buttons__Button-dz1ezt-0 dWQGoI\\">test</button>"`;
